File: LeetCode/203. Remove Linked List Elements/Solution.java
 

class Solution {

    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }

    public ListNode removeElements(ListNode head, int val) {
        if(head==null){
            return null;
        }
        ListNode ans=new ListNode();
        ans.next = head;
        ListNode cur=new ListNode();
        cur.next = ans;        
        while(cur.next!=null){
            if(cur.next.val == val){
                cur.next = cur.next.next;
            }
            else{
                cur = cur.next;
            }
        }
        return ans.next;
    }
}

File: LeetCode/203. Remove Linked List Elements/Solution.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        ans = cur = ListNode(0, head)
        while cur.next:
            if cur.next.val == val:
                cur.next = cur.next.next
            else:
                cur = cur.next
        return ans.next